inline/src/lexer/token.rs

```diff
@@ -150,10 +144,21 @@ impl Token {
     pub fn is_nesting_token(&self) -> bool {
         !matches!(
             self.kind,
-            TokenKind::Plain | TokenKind::Newline | TokenKind::Whitespace | TokenKind::EndOfLine
+            TokenKind::Plain
+                | TokenKind::Newline
+                | TokenKind::ExplicitNewline
+                | TokenKind::Whitespace
+                | TokenKind::ExplicitWhitespace
         )
     }
 
+    /// Checks whether this [`Token`] represents plain whitespace content (i.e. space or tab).
+    ///
+    /// [`Token`]: self::Token
+    pub fn is_plain_whitespace(&self) -> bool {
+        matches!(self.kind, TokenKind::Whitespace)
```

Comment:
The idea of this function is to check whether a given token is a whitespace that can be consumed by a plain inline. This is important when we want to merge multiple whitespace tokens, or if a plain inline starts with that kind of token (implementation detail). Do you have recommendations for a better name? 

I will see if there's a way to solve this in a more generic way 🤔. Maybe the way to go is two functions: 

1. `consumable_by(&self, inline: &Inline) -> bool` which tests whether the given token (`&self`) can be consumed as content of the given inline
2. `can_merge_with(&self, other: &Token) -> bool` which tests whether the given token (`&self`) can merge with the `other` token. This would also enable merging of different kinds of tokens, make it more generic etc, but also complicate a little bit even though we don't have a need for that right now. 

What do you think?

Reply:
uh I like that 👍 
maybe add `consumable_by_plain(&self) -> bool`, which should nearly be a drop in replacement of `is_plain_whitespace`, right?

Reply:
Yes that is correct. I will introduce function you suggest, and later on if the need arises to check in a more generic way we can change it.